apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/kotlin/adapter/AdapterCommon.kt

```diff
@@ -56,6 +56,12 @@ internal fun readFromResponseCodeBlock(
     )
   }.joinToCode(separator = "\n", suffix = "\n")
 
+  val path = if (syntheticProperties.any { it.condition != BooleanExpression.True }) {
+    CodeBlock.of("val $__path = $reader.$getPath()")
```

Comment:
Can we refine the condition here and only call `getPath()` if we know we're going to need it (because there's a `BLabel` condition? This is a hot path and will save some parsing time for the vast majority of users that do not use `@defer`

Reply:
I think yes although, at the moment the path is always passed to `evaluate`. But we could make the `evaluate`'s `path` parameter a `String?`, and initialize it to `null` if not needed instead of not defining it at all - WDYT?

Reply:
> I think yes although, at the moment the path is always passed to evaluate. But we could make the evaluate's path parameter a String?, and initialize it to null if not needed instead of not defining it at all - WDYT?

Agreed, passing null is the way to go. Either go with `val __path = null` or pass `null` to `evaluate()` directly like when `__typename` is not needed